src/firewall/client.py

```diff
@@ -2700,6 +2700,9 @@ def addIPSet(self, name, settings):
 
     @handle_exceptions
     def getZoneNames(self):
+        return self.real_getZoneNames()
+
+    def real_getZoneNames(self):
```

Comment:
I don't see the value is implementing the `real_*()` functions outside of `real_get_property()`. Basically we need to know if firewalld is running. If it's not, then we dump "firewalld is not running" and we're done. If it is running, we can call the normal `get*()` functions.

Reply:
If think `@handle_exceptions` is overused and the handler just killing the application is unsuitable in certain cases.

If firewalld is not running, then the output currently would be:
```
State: not running
Version: unknown (client: 2.0.999)

usage: 'firewall-cmd --help' for usage information or see firewall-cmd(1) man page
```
Instead of "Error: org.freedesktop.DBus.Error.NameHasNoOwner: firewalld is not running" from the handler.


You might think, just check for "NOT_RUNNING", afterwards all accesses (like `self.getZoneNames()`) are safe. But they are not. Any subsequent call can still fail with a D-Bus exception, which would result the `@handle_exceptions` handler to kill the process.

Instead, by handling errors ourselves and using raw/real methods, the error behavior is determined by `show_state_and_exit()`.